### PR TITLE
Add bounds to monad-control in streamly.cabal

### PR DESCRIPTION
**Pull request by rnjtranjan**

### CODE REVIEW

File: stack.yaml [hunk 1]
```diff
@@ -9,6 +9,7 @@ extra-deps:
   - unicode-data-0.2.0
   - tasty-bench-0.3.1
   - lockfree-queue-0.2.4
+  - monad-control-1.0.3@sha256:16e320d0d7ee4cd88d9f73bf1bc9af46005f9eb121adab8856d43e57f9cd64aa,2363
```
Comment (harendra-kumar) [inline stack.yaml:12]: Remove.

File: core/src/Streamly/Internal/Data/Stream/Bottom.hs [hunk 2]
```diff
@@ -532,3 +532,14 @@ zipWithM f m1 m2 = fromStreamD $ D.zipWithM f (toStreamD m1) (toStreamD m2)
...
+repeatMWith :: (m a -> Stream m a -> Stream m a) -> m a -> Stream m a
+repeatMWith cns = go
+
+    where
+
+    go m = m `cns` go m
+
+{-# INLINE_EARLY repeatM #-}
+repeatM :: (Monad m) => m a -> Stream m a
+repeatM = repeatMWith consM
```
Comment (harendra-kumar) [inline core/src/Streamly/Internal/Data/Stream/Bottom.hs:545]: Put it in a separate PR.

**Review by harendra-kumar — CHANGES_REQUESTED** [review]

**Review by adithyaov — APPROVED** [review]
